draft-ietf-quic-qpack.md

```diff
@@ -369,60 +369,72 @@ name and same value).  Therefore, duplicate entries MUST NOT be treated as an
-cautioned to avoid deleting the referenced name if the referenced entry is
-evicted from the dynamic table prior to inserting the new entry.
+cautioned to avoid deleting the referenced name or value if the referenced entry
+is evicted from the dynamic table prior to inserting the new entry.
+
+Whenever the dynamic table capacity is reduced by the encoder, entries are
+evicted from the end of the dynamic table until the size of the dynamic table is
+less than or equal to the new table capacity.  This mechanism can be used to
+completely clear entries from the dynamic table by setting a capacity of 0,
+which can subsequently be restored.
+
 
-Whenever the maximum size for the dynamic table is reduced by the encoder,
-entries are evicted from the end of the dynamic table until the size of the
-dynamic table is less than or equal to the new maximum size.  This mechanism can
-be used to completely clear entries from the dynamic table by setting a maxiumum
-size of 0, which can subsequently be restored.
+### Maximum Dynamic Table Capacity
 
+To bound the memory requirements of the decoder, the value that the encoder is
+permitted to set as dynamic table capacity is limited.  In HTTP/3, this value is
+determined by the SETTINGS_MAXIMUM_DYNAMIC_TABLE_CAPACITY setting sent by the
+decoder (see {{configuration}}).  The encoder MUST not set a dynamic table
+capacity that exceeds this maximum, but it can choose to use a lower dynamic
+table capacity (see {{capacity-update}}).
 
-### Maximum Table Size
 
-The encoder decides how to update the dynamic table size and as such can control
-how much memory is used by the dynamic table.  To limit the memory requirements
-of the decoder, the dynamic table size is strictly bounded.  The decoder
-determines the maximum size that the encoder is permitted to set for the dynamic
-table.  In HTTP/3, this value is determined by the SETTINGS_HEADER_TABLE_SIZE
-setting (see {{configuration}}).  The encoder MUST not set a dynamic table size
-that exceeds this maximum, but it can choose to use a lower dynamic table size
-(see {{size-update}}).
+### Initial Dynamic Table Capacity
 
-The initial maximum size is determined by the corresponding setting when HTTP
-requests or responses are first permitted to be sent. For clients using 0-RTT
-data in HTTP/3, the table size is the remembered value of the setting, even if
-the server later specifies a larger maximum in its SETTINGS frame.  For HTTP/3
-servers and HTTP/3 clients when 0-RTT is not attempted or is rejected, the
-initial maximum table size is the value of the setting in the peer's SETTINGS
-frame.
+The initial dynamic table capacity is the value of the maximum dynamic table
+capacity when HTTP requests or responses are first permitted to be sent.  For
+HTTP/3 servers and HTTP/3 clients when 0-RTT is not attempted or is rejected,
+the initial maximum table capacity is zero.  For clients using 0-RTT data in
```

Comment:
I have changed the initial maximum table capacity to zero in the spirit of https://github.com/quicwg/base-drafts/pull/2038, because that is what's consistent with the first sentence.  And endpoint is allowed to send HTTP requests and responses before receiving the peer's SETTINGS frame.

Another option is to scratch the first sentence, and describe that an endpoint cannot use the dynamic table until the SETTINGS frame is received, however, at that point the dynamic table capacity is changed to the value of SETTINGS_MAXIMUM_DYNAMIC_TABLE_CAPACITY.

Reply:
Good catch!  I like your proposed text, and this reflects the agreement we reached on #1530.